tftpd/tftpd.c

```diff
@@ -1654,16 +1689,49 @@ static void tftp_sendfile(const struct formats *pf, struct tftphdr *oap, int oac
                 goto abort;
 
             if (ap_opcode == ACK) {
+                /* Whole window sent */
                 if (ap_block == block) {
+                    offset = window = 0;
+                    timeout = rexmtval;
+                    break;
+                }
+
+                if ((ap_block < block && block - ap_block < windowsize) ||
```

Comment:
To make this a bit more readable, I would negate this check and jump to to "fail section in the end"
if(ap obviously wrong)
goto fail;

process the correct ap;

fail:
synchnet......